### PR TITLE
Bug 1566545 - Card UI spoc variants

### PR DESCRIPTION
**Pull request by punamdahiya**
Pre-req:
- Direct response promo/spoc with cta property set returned by Adzerk server https://spocs.getpocket.com/spocs

QA steps to test Link CTA variant
- By default a promo or spoc item should display CTA as link

QA steps to test Button CTA variant
- Set browser.newtabpage.activity-stream.discoverystream.config  "hardcode_layout" as false and layout_endpoint as 
"https://gist.githubusercontent.com/punamdahiya/da9f68852df22117c027c70eb99f3dbe/raw/f396f088233d981f39f843e4e1af9824d7165314/cta-button-4-spocs.json"

- Set browser.newtabpage.activity-stream.discoverystream.endpoints to https://,http://

A promo or spoc item should display CTA as button

### CODE REVIEW

File: content-src/components/DiscoveryStreamComponents/DSCard/DSCard.jsx [hunk 1]
```diff
@@ -9,6 +9,34 @@ import { ImpressionStats } from "../../DiscoveryStreamImpressionStats/Impression
...
+  <div className="meta">
+    <div className="info-wrap">
+      <p className="source clamp">
+        {source}
+        {context && <span className="context top">&#8226; Sponsored</span>}
+      </p>
+      <header className="title clamp">{title}</header>
+      {excerpt && <p className="excerpt clamp">{excerpt}</p>}
+      {isPromo && (
+        <button className="button learn-more-button">Learn More</button>
```
Comment (ScottDowne) [inline content-src/components/DiscoveryStreamComponents/DSCard/DSCard.jsx]: Is the button tied to being a promo, or is just just a button we can use for any cta/link?

Comment (punamdahiya) [reply content-src/components/DiscoveryStreamComponents/DSCard/DSCard.jsx]: Its tied to being a promo. We don't show button for publisher content in card. Clarified with @wolasi button is linked to same url as the card_url

**Comment by punamdahiya** [comment]
Patch display all promos in CardGrid if campaign_id/promo_id is set against CardGrid. A promo will paid promo if context ("Sponsored by...") is set against an item.

**Comment by punamdahiya** [comment]
@wolasi sharing links with cta-button and cta-link promo variants for design review. Thanks

https://drive.google.com/file/d/1Gxlvxxx5JZLqQYBNR5rDtLT1BKFjuBUc/view?usp=sharing
https://drive.google.com/file/d/1cb4UFGHf3crLNcf2b5SW2Hye1RKLz_0f/view?usp=sharing

**Comment by wolasi** [comment]
@punamdahiya the bottom padding for the button variant is wrong. should be 16px

**Comment by punamdahiya** [comment]
@wolasi Sharing link for cta-link and cta-button in default and dark theme for design review. As discussed in bug in 70 we will not be supporting CTA inside Hero component. Thanks

[CTA link default theme](https://drive.google.com/file/d/1XRMH1jBOVswCvAGnCcQztwLzhgnVUyf2/view?usp=sharing
)

[CTA link dark theme](https://drive.google.com/file/d/1PJX7LHwbSg8zU4AqGlMJYJvQmERnQ3C1/view?usp=sharing)

[CTA button default theme](https://drive.google.com/file/d/1w4nBacHh9q-szlBEyNWPSx2-e6j5Jkr0/view?usp=sharing)

[CTA button dark theme](https://drive.google.com/file/d/1ydYOoOLlQ2-4Te7-F3mF8mJzk2x-UNZB/view?usp=sharing)

**Comment by wolasi** [comment]
@punamdahiya any chance i could review in a build as described here https://docs.google.com/document/d/1GdOQR90hDcdaqXfcumhA-Vi_V3bbpyVjdiEimKQqRQc/edit#

**Comment by punamdahiya** [comment]
> @punamdahiya any chance i could review in a build as described here https://docs.google.com/document/d/1GdOQR90hDcdaqXfcumhA-Vi_V3bbpyVjdiEimKQqRQc/edit#

sure, I will put out build for cta-link and cta-button either with test spoc endpoint from mathijs or hardcoded whichever is earlier by EOD today

**Comment by wolasi** [comment]
thanks!

**Comment by wolasi** [comment]
Design feedback based on builds…

**CTA Link**
* Reduce spacing between text and caret in CTA link, give `.cta-link` `padding-right: 9px`
* Let's use the following markup instead
`<p><a href="#" class="cta-link icon icon-arrow" tabindex="0">Learn more</a></p>`

**CTA Button**
* make `.ds-card .meta .cta-button` `margin: 12px 0 4px;`
* use this `·` for the bullet in the `source · sponsored` lock up
* use spaces not padding in the `source · sponsored` lock up

**Pull request merged**